api/utils/requestProcessor.js

```diff
@@ -2148,6 +2202,132 @@ const validateAppForWriteAPI = (params, done, try_times) => {
+                    common.returnMessage(params, 400, 'Request does not match checksum');
+                    return done ? done() : false;
+                }
+            }
+            else {
+                console.log("Request does not have checksum", params.href, params.req.body);
+                common.returnMessage(params, 400, 'Request does not have checksum');
+                return done ? done() : false;
+            }
+        }
+
+        if (typeof params.qstring.tz !== 'undefined' && !isNaN(parseInt(params.qstring.tz))) {
+            params.user.tz = parseInt(params.qstring.tz);
+        }
+
+        if (params.qstring.metrics && typeof params.qstring.metrics === "string") {
+            try {
+                params.qstring.metrics = JSON.parse(params.qstring.metrics);
+            }
+            catch (SyntaxError) {
+                console.log('Parse metrics JSON failed for sdk fetch request', params.qstring.metrics, params.req.url, params.req.body);
+            }
+        }
+
+        Promise.all([fetchAppUser(params), countlyApi.data.usage.setLocation(params)]).then(() => {
+            if (params.qstring.metrics) {
+                try {
+                    countlyApi.data.usage.returnAllProcessedMetrics(params);
+                }
+                catch (ex) {
+                    console.log("Could not process metrics");
+                }
+            }
+
+            processFetchRequest(params, app, done);
+        }).catch(() => {
+            if (params.qstring.metrics) {
+                try {
+                    countlyApi.data.usage.returnAllProcessedMetrics(params);
+                }
+                catch (ex) {
+                    console.log("Could not process metrics");
+                }
+            }
+
+            processFetchRequest(params, app, done);
+        });
+    });
+};
+
+/**
+ * Function to fetch app user from db
+ * @param  {object} params - params object
+ * @returns {promise} - user
+ */
+const fetchAppUser = (params) => {
+    return new Promise((resolve) => {
+        common.db.collection('app_users' + params.app_id).findOne({'_id': params.app_user_id}, (err2, user) => {
+            params.app_user = user || {};
+            return resolve(user);
+        });
+    });
+};
+
+/**
+ * Add devices to ignore them
+ * @param  {params} params - params object
+ * @param  {function} done - callback when processing done
+ * @returns {function} done
+ */
+const ignorePossibleDevices = (params, done) => {
+    //ignore possible opted out users for ios 10
+    if (params.qstring.device_id === "00000000-0000-0000-0000-000000000000") {
+        common.returnMessage(params, 400, 'Ignoring device_id');
+        common.log("request").i('Request ignored: Ignoring zero IDFA device_id', params.req.url, params.req.body);
+        params.cancelRequest = "Ignoring zero IDFA device_id";
+        plugins.dispatch("/sdk/cancel", {params: params});
```

Comment:
If IDFA is 0, this event will log request in logger, 
I think we have some flag if log was already processed or not for request, but just noting :)